### PR TITLE
Dnahovskyi/grok 11348 ci build dockerfiles in GitHub actions

### PR DESCRIPTION
**Pull request by ksumag**

### CODE REVIEW

File: .github/workflows/packages.yml [hunk 1]
```diff
@@ -23,7 +23,7 @@ jobs:
       - name: Checkout
         uses: actions/checkout@v3
         with:
-          fetch-depth: 3
+          fetch-depth: 3
```
Comment (dnillovna) [inline .github/workflows/packages.yml:26]: Could you remove trailing spaces?

File: .github/workflows/packages.yml [hunk 2]
```diff
@@ -406,3 +464,8 @@ jobs:
             done
             echo "::set-output name=push::yes"
           fi
+
+
+
+
+
```
Comment (dnillovna) [inline .github/workflows/packages.yml:471]: Could you remove the new lines at the end? Leave just one line

File: .github/workflows/packages.yml [hunk 3]
```diff
@@ -230,6 +230,64 @@ jobs:
...
+        uses: andstor/file-existence-action@v2
+        with:
+          files: "dockerfiles"  
+      - name: check if dockerfies directory not empty 
+        id: NEdockerdir
+        run: |
+          if [ -z "$(ls -A  packages/${{ matrix.package }}/dockerfiles)" ]; then
+          echo "DIR_EMPTY=true" >> $GITHUB_ENV
+          else
+          echo "DIR_EMPTY=false" >> $GITHUB_ENV
```
Comment (dnillovna) [inline .github/workflows/packages.yml:245]: Could you use set outputs instead of environment variables?

Comment (dnillovna) [reply .github/workflows/packages.yml:245]: https://docs.github.com/en/actions/using-workflows/workflow-commands-for-github-actions#setting-an-output-parameter

**Review by dnillovna — CHANGES_REQUESTED** [review]

File: .github/workflows/packages.yml [hunk 4]
```diff
@@ -230,6 +230,64 @@ jobs:
...
+        run: |
+          echo "IMAGE_VER=`echo "$(jq .version package.json | sed -E -e 's/(^"|"$)//g')"`" >> $GITHUB_ENV
+          echo "PACKAGE_NAME=`echo "$(jq .name package.json | sed -E -e 's/(^"|"$)//g' -e 's,@datagrok/,,g')"`" >> $GITHUB_ENV
+
+      - name: Build and push package dockerimages
+        if: steps.check_directory.outputs.files_exists == 'true' && ${{env.DIR_EMPTY}} == 'false' && steps.publish.outcome == 'success'
+        uses: docker/build-push-action@v3
+        with:
+          context: packages/${{ matrix.package }}/dockerfiles
+          push: true
```
Comment (dnillovna) [inline .github/workflows/packages.yml:271]: Use condition `${{ }}` instead of two actions

File: .github/workflows/packages.yml [hunk 5]
```diff
@@ -230,6 +230,64 @@ jobs:
...
+        uses: docker/login-action@v2
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }} 
+        if: steps.check_directory.outputs.files_exists == 'true' && ${{env.DIR_EMPTY}} == 'false' && steps.publish.outcome == 'success'
+
+      - name: get docker image version
+        working-directory: packages/${{ matrix.package }}
+        run: |
+          echo "IMAGE_VER=`echo "$(jq .version package.json | sed -E -e 's/(^"|"$)//g')"`" >> $GITHUB_ENV
```
Comment (dnillovna) [inline .github/workflows/packages.yml:263]: Add commit ID to image version if we do not plan to publish image to Docker Hub

Comment (dnillovna) [inline .github/workflows/packages.yml:263]: Instead of sed use `-r` flag for jq all over the file

File: .github/workflows/packages.yml [hunk 6]
```diff
@@ -230,6 +230,64 @@ jobs:
...
+        uses: docker/build-push-action@v3
+        with:
+          context: packages/${{ matrix.package }}/dockerfiles
+          push: true
+          tags: datagrok/${{env.PACKAGE_NAME}}:${{env.IMAGE_VER}}
+        
+
+    
+      - name: build dockerfiles as artifact when npm push failed
+        if: steps.directory_changes_checker.output.hasChanges == 'true' && steps.publish.outcome == 'failure'
```
Comment (dnillovna) [inline .github/workflows/packages.yml:277]: Check that `steps.publish.outcome` is skipped or a failure

File: .github/workflows/packages.yml [hunk 7]
```diff
@@ -230,6 +230,64 @@ jobs:
...
+        uses: docker/build-push-action@v3
+        with:
+          context: packages/${{ matrix.package }}/dockerfiles
+          push: false
+          tags: datagrok/${{env.PACKAGE_NAME}}:${{env.IMAGE_VER}}
+        
+      - name: Upload dockerfiles as artifact when npm push failed
+        if: steps.directory_changes_checker.output.hasChanges == 'true' && steps.publish.outcome == 'failure'
```
Comment (dnillovna) [inline .github/workflows/packages.yml:285]: Check that steps.publish.outcome is skipped or a failure

File: .github/workflows/packages.yml [hunk 8]
```diff
@@ -230,6 +230,64 @@ jobs:
...
+          directory: "packages/${{ matrix.package }}/dockerfiles"
+      - name: login to DockerHub   
+        uses: docker/login-action@v2
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }} 
+        if: steps.check_directory.outputs.files_exists == 'true' && ${{env.DIR_EMPTY}} == 'false' && steps.publish.outcome == 'success'
+
+      - name: get docker image version
+        working-directory: packages/${{ matrix.package }}
```
[the remark this answers is not in the document]
Comment (ksumag) [reply .github/workflows/packages.yml:261]: no, we needs it anyway - to tag an image after building image and for upload image artefact

**Pull request closed without merging**